server/middleware/populateSelectedEstablishment.ts

```diff
@@ -1,15 +1,28 @@
 import type { RequestHandler } from 'express'
 import asyncMiddleware from './asyncMiddleware'
 import { Services } from '../services'
+import logger from '../../logger'
 
 export default function populateSelectedEstablishment({ supportedPrisonsService }: Services): RequestHandler {
   return asyncMiddleware(async (req, res, next) => {
-    if (req.session.selectedEstablishment === undefined && req.path !== '/change-establishment') {
-      const supportedPrisons = await supportedPrisonsService.getSupportedPrisons(res.locals.user.username)
+    if (req.path === '/change-establishment') {
```

Comment:
Updated this middleware for interim step of using DPS header but still maintaining our own 'Change establishment' page. This is being removed next so this code will get much simpler.